siliconcompiler/tools/yosys/syn_fpga.tcl

```diff
@@ -54,7 +168,13 @@ if {[string match {ice*} $sc_partname]} {
     #Do our own thing from here
     yosys techmap -map +/techmap.v
 
-    yosys dfflegalize -cell \$_DFF_P_ 01 -cell \$_DFF_P??_ 01
+    if { $sc_syn_legalize_flops != 0 } {
+        legalize_flops \
+            $sc_syn_flop_async_reset \
+            $sc_syn_flop_async_set \
+            $sc_syn_flop_enable \
+            $sc_syn_flop_library
+    }
```

Comment:
probably something like:
```
if { [llength $sc_ff_features] != 0 } {
...
}
```

Reply:
The purpose of having a boolean flag to allow/disallow flop legalization is to provide users a backdoor to skip this step for debugging purposes; in general, the step should always be executed in a final flow.